fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/logical/PushDownNotSlotReferenceExpressionOfOnClause.java

```diff
@@ -0,0 +1,200 @@
+                                return new LogicalProject<>(slots, p.getGroup().getLogicalExpression().getPlan());
+                            })
+                            .collect(Collectors.toList());
+
+                    LogicalJoin<Plan, Plan> joinAddProjects = (LogicalJoin<Plan, Plan>) join.withChildren(childs);
+
+                    return joinAddProjects
+                            .withCondition(Optional.of(join.getCondition().get().accept(
+                                    new TreeReplaceVisitor(),
+                                    exprMap.values().stream()
+                                            .reduce(new HashMap<>(), (map1, map2) -> {
+                                                map1.putAll(map2);
+                                                return map1;
+                                            })
+                            ))
+                    );
+                }).toRule(RuleType.PUSH_DOWN_NOT_SLOT_REFERENCE_EXPRESSION);
+    }
+
+    private static class TreeCollectVisitor extends ExpressionVisitor<Expression,
+            Map<SlotReference, Map<Expression, Expression>>> {
+
+        @Override
+        public Expression visit(Expression expr,
+                Map<SlotReference, Map<Expression, Expression>> context) {
+            Set<SlotReference> set = new HashSet<>(expr.collect(SlotReference.class::isInstance));
+            if (set.size() == 1) {
+                SlotReference ref = set.stream().findFirst().get();
+                Map<Expression, Expression> map = context.get(ref);
+                if (map == null) {
+                    Map<Expression, Expression> newMap = new HashMap<>();
+                    newMap.put(expr, new Alias(expr, expr.toString()));
+                    context.put(ref, newMap);
+                } else {
+                    map.put(expr, new Alias(expr, expr.toString()));
+                }
+            }
+            return expr;
+        }
+
+        public Expression visitChildren(Expression expr,
+                Map<SlotReference, Map<Expression, Expression>> ctx) {
+            expr.children().forEach(node -> node.accept(this, ctx));
+            return expr;
+        }
+
+        @Override
+        public Expression visitComparisonPredicate(ComparisonPredicate predicate,
+                Map<SlotReference, Map<Expression, Expression>> ctx) {
+            return visitChildren(predicate, ctx);
+        }
+
+        @Override
+        public Expression visitCompoundPredicate(CompoundPredicate predicate,
+                Map<SlotReference, Map<Expression, Expression>> ctx) {
+            return visitChildren(predicate, ctx);
+        }
+    }
+
+    private static class TreeReplaceVisitor extends ExpressionVisitor<Expression, Map<Expression, Expression>> {
+
+        @Override
+        public Expression visit(Expression expr, Map<Expression, Expression> context) {
+            expr.children().forEach(node -> node.accept(this, context));
```

Comment:
if ur default visit function do not return expression with new children. u can not get correct expression at all. btw u could refer to default expression rewriter